hstrat/serialization/_unpack_differentiae.py

```diff
@@ -10,23 +11,25 @@ def unpack_differentiae(
     packed_differentiae: str,
     differentia_bit_width: int,
 ) -> typing.Iterable[int]:
-    """Unpack a sequence of differentiae from into a compact, concatenated
-    base 64 representation.
+    """Unpack a compact, concatenated base 64 representation into
+    a sequence of differentiae.
     """
     _bytes = b64decode(packed_differentiae)
-    bits = BitArray(bytes=_bytes)
 
-    # if padding bits are possible, first byte tells how many are required
-    if differentia_bit_width % 8:
+    if differentia_bit_width in [8, 16, 32, 64]:
+        # bit width is a multiple of 8 -- no padding required
+        dt = np.dtype(f">u{differentia_bit_width // 8}")
+        yield from np.frombuffer(_bytes, dtype=dt)
+    else:
+        # padding bits are possible, first byte tells how many are required
+        bits = BitArray(bytes=_bytes)
         num_padding_bits = bits[:8].uint
         if num_padding_bits:
             valid_bits = bits[8:-num_padding_bits]
         else:
             valid_bits = bits[8:]
-    else:
-        valid_bits = bits[:]
 
-    assert len(valid_bits) % differentia_bit_width == 0
+        assert len(valid_bits) % differentia_bit_width == 0
 
-    for chunk in iter_chunks(valid_bits, differentia_bit_width):
-        yield chunk.uint
+        for chunk in iter_chunks(valid_bits, differentia_bit_width):
+            yield chunk.uint
```

Comment:
Split out the bitarray and numpy implementations into sep functions  within same file private leading underscore

Reply:
Done in fb47b915